app/gui2/src/util/ast/node.ts

```diff
@@ -1,19 +1,28 @@
 import type { Node } from '@/stores/graph'
 import { Ast } from '@/util/ast'
+import { Prefixes } from '@/util/ast/prefixes'
 import { Vec2 } from '@/util/data/vec2'
 
+export const prefixes = Prefixes.FromLines({
+  enableRecording:
+    'Standard.Base.Runtime.with_enabled_context Standard.Base.Runtime.Context.Output __ <| __',
+})
+
 export function nodeFromAst(ast: Ast.Ast): Node | undefined {
   const nodeCode = ast instanceof Ast.Documented ? ast.expression : ast
   if (!nodeCode) return
   const pattern = nodeCode instanceof Ast.Assignment ? nodeCode.pattern : undefined
   const rootSpan = nodeCode instanceof Ast.Assignment ? nodeCode.expression : nodeCode
+  const { innerExpr, matches } = prefixes.extractMatches(rootSpan)
```

Comment:
@kazcw would you happen to know why this errors with this error on the second click?
```
Uncaught Error: Assertion failed: id in module
    bail assert.ts:68
    assert assert.ts:6
    get mutableModule.ts:253
    children tree.ts:146
    zipLongest iterable.ts:54
    matchSubtree match.ts:75
    match match.ts:27
    matches prefixes.ts:30
    extractMatches prefixes.ts:29
    nodeFromAst node.ts:16
    readFunctionAst graphDatabase.ts:347
    updateState index.ts:146
    handleModuleUpdate index.ts:127
    handle index.ts:94
```

Reply:
or perhaps it's erroring on `mutableModule`s only? perhaps due to some inconsistency in the state of the tree?

Reply:
The problem is caused by mutating data returned from Y.Js. The current `develop` changes access patterns and happens to fix the case encountered in this PR. I'll open a PR soon addressing the root cause.